app/src/main/java/de/westnordost/streetcomplete/quests/oneway_suspects/data/TrafficFlowSegmentsApi.kt

```diff
@@ -2,24 +2,31 @@ package de.westnordost.streetcomplete.quests.oneway_suspects.data
 
 import de.westnordost.streetcomplete.data.osm.mapdata.BoundingBox
 import de.westnordost.streetcomplete.util.ktx.format
+import io.ktor.client.HttpClient
+import io.ktor.client.call.body
+import io.ktor.client.plugins.expectSuccess
+import io.ktor.client.request.get
 import kotlinx.serialization.Serializable
 import kotlinx.serialization.json.Json
-import java.net.URL
 
-/** Dao for using this API: https://github.com/ENT8R/oneway-data-api  */
-class TrafficFlowSegmentsApi(private val apiUrl: String) {
+/** Dao for using this API: https://github.com/streetcomplete/oneway-data-api  */
+class TrafficFlowSegmentsApi(
+    private val httpClient: HttpClient,
+    private val apiUrl: String
+) {
 
-    fun get(bbox: BoundingBox): Map<Long, List<TrafficFlowSegment>> {
+    suspend fun get(bbox: BoundingBox): Map<Long, List<TrafficFlowSegment>> {
         val leftBottomRightTopString = listOf(
             bbox.min.longitude,
             bbox.min.latitude,
             bbox.max.longitude,
             bbox.max.latitude
         ).joinToString(",") { it.format(7) }
 
-        val url = URL("$apiUrl?bbox=$leftBottomRightTopString")
-        val json = url.openConnection().getInputStream().bufferedReader().use { it.readText() }
-        return parse(json)
+        val response = httpClient.get("$apiUrl?bbox=$leftBottomRightTopString") {
+            expectSuccess = true
```

Comment:
So what happens when it is not successful? Does it throw an exception? [The documentation is really not that helpful.](https://api.ktor.io/ktor-client/ktor-client-core/io.ktor.client.plugins/expect-success.html)

Reply:
Good question, so `expectSucess` will throw a [few different types of exception](https://github.com/ktorio/ktor/blob/2.3.8/ktor-client/ktor-client-core/common/src/io/ktor/client/plugins/DefaultResponseValidation.kt#L53-L56). I've added some test cases to demonstrate `ClientRequestException` and `ServerResponseException`.